app/namespace.go

```diff
@@ -138,179 +182,215 @@ func NewNamespaceCommand(getNamespaceClientFn GetNamespaceClientFn) (CommandOut,
+					if err != nil {
+						return err
+					}
+					n, err := c.getNamespace(ctx.String(NamespaceFlagName))
+					if err != nil {
+						return err
+					}
+					existingCerts, err := parseCertificates(n.Spec.AcceptedClientCa)
+					if err != nil {
+						return err
+					}
+					err = existingCerts.remove(removeCerts)
+					if err != nil {
+						return err
+					}
+					bundle, err := existingCerts.bundle()
+					if err != nil {
+						return err
+					}
+					if n.Spec.AcceptedClientCa == bundle {
+						return errors.New("nothing to change")
+					}
+					n.Spec.AcceptedClientCa = bundle
+					return c.updateNamespace(ctx, n)
+				},
+			}, {
+
+				Name:  "set",
+				Usage: "set the accepted client ca certificate",
+				Flags: []cli.Flag{
+					NamespaceFlag,
+					RequestIDFlag,
+					ResourceVersionFlag,
+					caCertificateFlag,
+					caCertificateFileFlag,
+				},
+				Action: func(ctx *cli.Context) error {
+					cert, err := readCACerts(ctx)
+					if err != nil {
+						return err
+					}
+					n, err := c.getNamespace(ctx.String(NamespaceFlagName))
+					if err != nil {
+						return err
+					}
+					if n.Spec.AcceptedClientCa == cert {
+						return errors.New("nothing to change")
+					}
+					n.Spec.AcceptedClientCa = cert
+					return c.updateNamespace(ctx, n)
+				},
+			}},
+		}, {
+			Name:    "search-attributes",
+			Usage:   "manage search attributes used by namespace",
+			Aliases: []string{"sa"},
+			Subcommands: []*cli.Command{{
+				Name:    "add",
+				Usage:   "add a new namespace custom search attribute",
+				Aliases: []string{"a"},
+				Flags: []cli.Flag{
+					NamespaceFlag,
+					RequestIDFlag,
+					ResourceVersionFlag,
+					&cli.StringSliceFlag{
+						Name:     "search-attribute",
+						Usage:    fmt.Sprintf("flag can be used multiple times; value must be \"name=type\"; valid types are: %v", getSearchAttributeTypes()),
+						Aliases:  []string{"sa"},
+						Required: true,
 					},
-					Action: func(ctx *cli.Context) error {
-						csa, err := toSearchAttributes(ctx.StringSlice("search-attribute"))
-						if err != nil {
-							return err
-						}
-						n, err := c.getNamespace(
-							ctx.String(NamespaceFlagName),
-						)
-						if err != nil {
-							return err
-						}
-						if n.State != ns.STATE_ACTIVE {
-							return fmt.Errorf("namespace not in '%s' state to perform update, current_state='%s'",
-								ns.Namespace_State_name[int32(ns.STATE_ACTIVE)],
-								ns.Namespace_State_name[int32(n.State)])
+				},
+				Action: func(ctx *cli.Context) error {
+					csa, err := toSearchAttributes(ctx.StringSlice("search-attribute"))
+					if err != nil {
+						return err
+					}
+					n, err := c.getNamespace(ctx.String(NamespaceFlagName))
+					if err != nil {
+						return err
+					}
+					if n.Spec.SearchAttributes == nil {
+						n.Spec.SearchAttributes = make(map[string]namespace.NamespaceSpec_SearchAttributeType)
+					}
+					for attrName, attrType := range csa {
```

Comment:
same question here: can we rely on the server

Reply:
we can't, add is just a cli side concept. it relies on the update namespace op.